server/service/integration_core_test.go

```diff
@@ -6360,6 +6360,27 @@ func (s *integrationTestSuite) TestOrbitConfigNotifications() {
 	require.False(t, resp.Notifications.RenewEnrollmentProfile)
 }
 
+func (s *integrationTestSuite) TestTryingToEnrollWithTheWrongSecret() {
+	t := s.T()
+	ctx := context.Background()
+
+	h, err := s.ds.NewHost(ctx, &fleet.Host{
+		HardwareSerial:   uuid.New().String(),
+		Platform:         "darwin",
+		LastEnrolledAt:   time.Now(),
+		DetailUpdatedAt:  time.Now(),
+		RefetchRequested: true,
+	})
+	require.NoError(t, err)
+
+	var resp EnrollOrbitResponse
+	s.DoJSON("POST", "/api/fleet/orbit/enroll", EnrollOrbitRequest{
+		EnrollSecret:   uuid.New().String(),
+		HardwareUUID:   h.UUID,
+		HardwareSerial: h.HardwareSerial,
+	}, http.StatusUnauthorized, &resp)
```

Comment:
Should we not use require here?
`require.Error(t, err)`

Reply:
Thanks - addressed in https://github.com/fleetdm/fleet/pull/11121/commits/5a16bd6a9999fb3178084e9bcbb4fbad599b7b10